src/extract-cli.js

```diff
@@ -67,7 +67,13 @@ files.forEach(function(filename) {
   try {
     let data = fs.readFileSync(file, {encoding: 'utf-8'}).toString();
     extractor.parse(file, extract.preprocessTemplate(data, ext));
-    extractor.parseJavascript(file, extract.preprocessJavascript(data, ext));
+    let script = extract.preprocessJavascript(data, ext)
+
+    if (script && script.lang === 'ts' ) {
+      extractor.parseTypeScript(file, script.content);
```

Comment:
I'd be really glad to see you using the same type as the function below, say: `script` and not `script.content`.

If thos two functions look like they have the same signature but they don't it'll lead to bad issues later.

Reply:
I am using `script.content` in both cases now. I changed `preprocessJavascript` earlier to return an object that contains `content` and `lang`. The other option would be to make `parseTypeScript` and `parseJavascript` handle this object, but I think that would make those functions less general.